### PR TITLE
fix: advertised address should fall back to configured address

### LINKED ISSUE #9658: Job activation delayed by 10 seconds under stable network conditions
**Describe the bug**

Sometimes jobs are activated exactly 10 seconds after the corresponding service task is activated. In general this can happen for various reasons, for example connection issue between worker, gateway and broker. 10 seconds is also the default request timeout for clients. 

We've recently started to observe this in our QA runs, where these delays cause test failures. Since the network should be in a stable condition and this has only recently become an issue, it seems likely that this is actually a regression that we need to fix.

**To Reproduce**

With a current snapshot version of Zeebe and a spring-zeebe `8.0.5` this was at least not trivially reproducible.

**Expected behavior**

Under stable network conditions and barring any other issues, there should be virtually no delay between service task activation and job activation.

**Log/Stacktrace**

```json
{
  "partitionId": 1,
  "value": {
    "version": 1,
    "processDefinitionKey": 2251799813685372,
    "bpmnProcessId": "sequential-test-process",
    "processInstanceKey": 4503599627370626,
    "parentElementInstanceKey": -1,
    "flowScopeKey": 4503599627370626,
    "parentProcessInstanceKey": -1,
    "elementId": "step-0",
    "bpmnElementType": "SERVICE_TASK"
  },
  "key": 4503599627370633,
  "timestamp": 1656635590315,
  "position": 393,
  "valueType": "PROCESS_INSTANCE",
  "intent": "ELEMENT_ACTIVATED",
  "recordType": "EVENT",
  "rejectionType": "NULL_VAL",
  "rejectionReason": "",
  "brokerVersion": "8.1.0",
  "sourceRecordPosition": 390
}
{
  "index": 176,
  "term": 1,
  "entries": [
    {
      "partitionId": 1,
      "value": {
        "type": "test-job",
        "worker": "default",
        "truncated": false,
        "timeout": 10000,
        "maxJobsToActivate": 32,
        "jobKeys": [],
        "jobs": []
      },
      "key": -1,
      "timestamp": 1656635600418,
      "position": 394,
      "valueType": "JOB_BATCH",
      "intent": "ACTIVATE",
      "recordType": "COMMAND",
      "rejectionType": "NULL_VAL",
      "rejectionReason": "",
      "brokerVersion": "8.1.0",
      "sourceRecordPosition": -1
    }
  ]
}
```

**Environment:**
- OS: 
- Zeebe Version: 
- Configuration:

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

Since we introduced support for advertised addresses in the gateway in #9572, the atomix communication services were bound to the default host and port (0.0.0.0:26502) if no advertised address was configured instead of falling back to the configured host and port first. 

## Related issues

closes #9658 because job available notifications were not received which caused the delay in job activation.

### CODE REVIEW

**Review by ChrisKujawa — COMMENTED** [review]
Thanks Ole 👍

I would like to have an integration test which fails without the fix actually, and/or that we use the StandaloneVateway setup code in our QA IT ClusterRule. Not sure whether we want to do that as follow up but it would be good to have it.

**Comment by oleschoenburg** [comment]
I agree, but I'd like to do this as a follow up since I'd like to include this fix in the next alpha release. See https://github.com/camunda/zeebe/issues/9670

**Comment by Zelldon** [comment]
I created an integration test (regression test), which fails with SNAPSHOT and is green with `gcr.io/zeebe-io/zeebe:8.1.0-SNAPSHOT-os-debug-longpolling-notif-b51f3971`. Please have a look whether this is fine with you. I hope it was ok for you that I did that.

**Review by ChrisKujawa — APPROVED** [review]

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/LongPollingIT.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    // given
+    cluster =
+        ZeebeCluster.builder()
+            .withBrokersCount(1)
+            .withGatewaysCount(1)
+            .withEmbeddedGateway(false)
+            .withPartitionsCount(3)
+            .withReplicationFactor(1)
+            .withBrokerImage(ZeebeTestContainerDefaults.defaultTestImage())
+            .withGatewayImage(ZeebeTestContainerDefaults.defaultTestImage())
```
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/LongPollingIT.java:73]:
```suggestion
            .withImage(ZeebeTestContainerDefaults.defaultTestImage())
```

**Pull request merged**